src/vcpkg/base/files.cpp

```diff
@@ -46,6 +80,365 @@ namespace
         Checks::exit_with_message(
             li, Strings::concat(call_name, "(", Strings::join(", ", args.begin(), args.end()), "): ", ec.message()));
     }
+
+    stdfs::copy_options convert_copy_options(CopyOptions options)
+    {
+        stdfs::copy_options result{};
+        const auto unpacked = static_cast<int>(options);
+
+        switch (unpacked & static_cast<int>(CopyOptions::existing_mask))
+        {
+            case static_cast<int>(CopyOptions::skip_existing): result |= stdfs::copy_options::skip_existing; break;
+            case static_cast<int>(CopyOptions::overwrite_existing):
+                result |= stdfs::copy_options::overwrite_existing;
+                break;
+            case static_cast<int>(CopyOptions::update_existing): result |= stdfs::copy_options::update_existing; break;
+        }
+
+        if (unpacked & static_cast<int>(CopyOptions::recursive))
+        {
+            result |= stdfs::copy_options::recursive;
+        }
+
+        switch (unpacked & static_cast<int>(CopyOptions::symlinks_mask))
+        {
+            case static_cast<int>(CopyOptions::copy_symlinks): result |= stdfs::copy_options::copy_symlinks; break;
+            case static_cast<int>(CopyOptions::skip_symlinks): result |= stdfs::copy_options::skip_symlinks; break;
+        }
+
+        return result;
+    }
+
+    FileType convert_file_type(stdfs::file_type type) noexcept
+    {
+        switch (type)
+        {
+            case stdfs::file_type::none: return FileType::none;
+            case stdfs::file_type::not_found: return FileType::not_found;
+            case stdfs::file_type::regular: return FileType::regular;
+            case stdfs::file_type::directory: return FileType::directory;
+            case stdfs::file_type::symlink: return FileType::symlink;
+            case stdfs::file_type::block: return FileType::block;
+            case stdfs::file_type::character: return FileType::character;
+            case stdfs::file_type::fifo: return FileType::fifo;
+            case stdfs::file_type::socket: return FileType::socket;
+            case stdfs::file_type::unknown: return FileType::unknown;
+#if defined(_WIN32)
+            case stdfs::file_type::junction: return FileType::junction;
+#endif // _WIN32
+            default: Checks::unreachable(VCPKG_LINE_INFO);
+        }
+    }
+
+    stdfs::path to_stdfs_path(const Path& utfpath)
+    {
+#if defined(_WIN32)
+        return stdfs::path(Strings::to_utf16(utfpath.native()));
```

Comment:
Is this better than `stdfs::u8path(utfpath.native())`?

Reply:
stdfs::u8path doesn't work because that string is chars, not char8_ts. (It's the same reason we needed `vcpkg::u8path` before)